bridge/client/app/app-header/app-header.component.ts

```diff
@@ -33,37 +32,41 @@ export class AppHeaderComponent implements OnInit, OnDestroy {
   ngOnInit() {
     this.projects = this.dataService.projects;
 
-    this.routeSub = this.router.events.subscribe(event => {
-      if(event instanceof RoutesRecognized) {
-        let projectName = event.state.root.children[0].params['projectName'];
-        this.project = this.dataService.projects.pipe(
-          filter(projects => !!projects),
-          map(projects => projects.find(p => {
-            return p.projectName === projectName;
-          }))
-        );
-      }
-    });
-
-    this.versionInfoSub = this.dataService.versionInfo.subscribe(versionInfo => {
-      this.versionInfo = versionInfo;
-      if(versionInfo.versionCheckEnabled === null) {
-        this.showVersionCheckInfoDialog();
-      } else if(versionInfo.versionCheckEnabled) {
-        if(semver.valid(versionInfo.keptnVersion)) {
-          if(versionInfo.availableVersions.cli)
-            this.doVersionCheck(versionInfo.keptnVersion, versionInfo.availableVersions.cli.stable, versionInfo.availableVersions.cli.prerelease, "Keptn");
-        } else {
-          versionInfo.keptnVersionInvalid = true;
+    this.router.events
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe(event => {
+        if(event instanceof RoutesRecognized) {
+          let projectName = event.state.root.children[0].params['projectName'];
+          this.project = this.dataService.projects.pipe(
+            filter(projects => !!projects),
+            map(projects => projects.find(p => {
+              return p.projectName === projectName;
+            }))
+          );
         }
-        if(semver.valid(versionInfo.bridgeVersion)) {
-          if(versionInfo.availableVersions.bridge)
-            this.doVersionCheck(versionInfo.bridgeVersion, versionInfo.availableVersions.bridge.stable, versionInfo.availableVersions.bridge.prerelease, "Keptn Bridge");
-        } else {
-          versionInfo.bridgeVersionInvalid = true;
+      });
+
+    this.dataService.versionInfo
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe(versionInfo => {
+        this.versionInfo = versionInfo;
+        if(versionInfo.versionCheckEnabled === null) {
+          this.showVersionCheckInfoDialog();
+        } else if(versionInfo.versionCheckEnabled) {
+          if(semver.valid(versionInfo.keptnVersion)) {
+            if(versionInfo.availableVersions.cli)
+              this.doVersionCheck(versionInfo.keptnVersion, versionInfo.availableVersions.cli.stable, versionInfo.availableVersions.cli.prerelease, "Keptn");
+          } else {
+            versionInfo.keptnVersionInvalid = true;
+          }
+          if(semver.valid(versionInfo.bridgeVersion)) {
+            if(versionInfo.availableVersions.bridge)
+              this.doVersionCheck(versionInfo.bridgeVersion, versionInfo.availableVersions.bridge.stable, versionInfo.availableVersions.bridge.prerelease, "Keptn Bridge");
+          } else {
+            versionInfo.bridgeVersionInvalid = true;
+          }
         }
```

Comment:
yep, actually the check if the version is semver.valid could be done inside `doVersionCheck` and that function could return `false` in case it is invalid.